filetracker/servers/migration.py

```diff
@@ -19,11 +20,12 @@ def __init__(self, redirect_url, dir=None):
         self.redirect_url = redirect_url
 
     def handle_redirect(self, environ, start_response, present_handler):
-        path = os.path.join(self.dir, self.get_path(environ))
-        if os.path.isfile(path):
+        endpoint, path = base.get_endpoint_and_path(environ)
+
+        if os.path.isfile(os.path.join(self.dir, path)):
             return present_handler(environ, start_response)
 
-        new_url = self.redirect_url + environ['PATH_INFO']
+        new_url = os.path.join(self.redirect_url, endpoint, path)
```

Comment:
This looks very suspicious - using function that is meant to manipulate OS paths to join URLs.
Maybe something like [urllib.parse.urljoin](https://docs.python.org/3/library/urllib.parse.html#urllib.parse.urljoin) could be used? (Though that function looks a bit strange)
Anyway, I think that using an OS function is worse than just concatenating the strings, because IIRC on Windows it would use back-slashes.